graylog2-server/src/main/java/org/graylog2/rest/resources/system/logs/LoggersResource.java

```diff
@@ -151,6 +146,43 @@ public void setSingleLoggerLevel(
         logger.setLevel(newLevel);
     }
 
+    @GET
+    @Timed
+    @ApiOperation(value = "Get recent log messages")
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "Memory appender not configured.")
+    })
+    @Path("/messages/recent")
+    @Produces(MediaType.APPLICATION_JSON)
+    public LogMessagesSummary messages(@ApiParam(name = "max", required = true) @QueryParam("max") int max) {
+        List<LogMessage> messages = new ArrayList<>();
+
+        Logger logger = Logger.getRootLogger();
+        if (logger.getAppender("memory") == null) {
+            throw new WebApplicationException(404);
```

Comment:
There's a `NotFoundException` which automatically sets the correct response status. Additionally, a short explaining message would be nice (e. g. "In-Memory appender not found. Please check the log4j configuration of this Graylog instance.")